src/com/ms/silverking/cloud/argus/TerminatorAsyncLogger.java

```diff
@@ -11,121 +11,121 @@
-                try {
-                    getLogger().log(Level.WARNING, terminatedProcMessage);
-                    handler.flush();
-                } catch (FileNotFoundException e){
-                    Log.logErrorWarning(e) ;
-                }
-
-            } catch (InterruptedException ie) {
-                continue;
-            }
-        }
-        Log.fine("TerminatorAsyncLogger stopping");
-    }
+  void add(String msg) {
+
+    logMessages.add(TimeUtils.getCurrentTimeString() + " : " + msg);
+  }
 
-    /**
-     * A method for testing TerminatorAsyncLogger
+  public void stopRunning() {
+    running.set(false);
+
+    /* asyncLogger is very likely to be blocked in logMessages.take()
+     * when no log messages in the queue, we can
+     * *either*  send an empty message *or* call interrupt()
+     * to re-read 'running' var and terminate messageHandler thread.
      */
-    public static void test() {
+    add("");
+    //asyncLogger.interrupt();
+  }
+
+  public void run() {
+    Log.fine("TerminatorAsyncLogger running");
+    while (running.get()) {
+      String terminatedProcMessage;
+
+      try {
+        terminatedProcMessage = logMessages.take();
         try {
-            Log.setLevelAll();
-            Log.setPrintStreams(System.out);  //main log
-
-            //create a temporary file name for async logger
-            File tmpFile = File.createTempFile("terminatorAsyncLogger.test", ".log");  
-            tmpFile.delete();  //remove the file, it should be created during lazy init. 
-            String fileName = tmpFile.getAbsolutePath();
-            Log.fine( "terminatorAsyncLogger log file: " + fileName);
-            
-            //create an object of this class
-            TerminatorAsyncLogger lmh = new TerminatorAsyncLogger(fileName);
-            
-            //add messages 
-            for(int i =0; i<20; i++) {
-              lmh.add("LogMessage Test" + i);
-            }
-            Thread.sleep(100);
-            lmh.stopRunning();
-            
-        } catch(Exception ex) {
-            System.err.println(ex);
+          getLogger().log(Level.WARNING, terminatedProcMessage);
+          handler.flush();
+        } catch (FileNotFoundException e) {
+          Log.logErrorWarning(e);
         }
+
+      } catch (InterruptedException ie) {
+        continue;
+      }
     }
-    
-    /**
-     * @param args cmd-line args
-     */
-    public static void main(String[] args) {
-        //run basic test, no arguments are used
-        test();
+    Log.fine("TerminatorAsyncLogger stopping");
+  }
+
+  /**
+   * A method for testing TerminatorAsyncLogger
+   */
+  public static void test() {
+    try {
+      Log.setLevelAll();
+      Log.setPrintStreams(System.out);  //main log
+
+      //create a temporary file name for async logger
+      File tmpFile = File.createTempFile("terminatorAsyncLogger.test", ".log");
+      tmpFile.delete();  //remove the file, it should be created during lazy init.
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The result of deleting a file on this line should be checked and failure should be logged. java.io.File.delete() will simply return false if it fails to delete the file, for example, when the file does not exist. It will not throw unless there is a security exception.